src/js/base/Pagination.jsx

```diff
@@ -36,33 +45,62 @@ const usePagination = (pageCount, currentPage) => {
     return paginationRange;
 };
 
-export const Pagination = ({ currentPage, pageCount, onPageChange }) => {
+export const Pagination = ({ documents, renderRow, currentPage, pageCount, onLoadNextPage }) => {
     const paginationRange = usePagination(pageCount, currentPage);
+    const location = useLocation();
+    const page = new URLSearchParams(location.search).get("page");
```

Comment:
If possible I would suggest making the pagination component agnostic of the source of method of setting the page numbers. On first blush I would suspect that the passed page number, onLoadNextPage, and page count should be able to handle most navigation